presto-raptorx/src/main/java/com/facebook/presto/raptorx/RaptorConnector.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.raptorx;
+
+import com.facebook.presto.raptorx.metadata.BucketManager;
+import com.facebook.presto.raptorx.metadata.NodeSupplier;
+import com.facebook.presto.raptorx.transaction.Transaction;
+import com.facebook.presto.raptorx.transaction.TransactionWriter;
+import com.facebook.presto.spi.SystemTable;
+import com.facebook.presto.spi.connector.Connector;
+import com.facebook.presto.spi.connector.ConnectorMetadata;
+import com.facebook.presto.spi.connector.ConnectorNodePartitioningProvider;
+import com.facebook.presto.spi.connector.ConnectorPageSinkProvider;
+import com.facebook.presto.spi.connector.ConnectorPageSourceProvider;
+import com.facebook.presto.spi.connector.ConnectorSplitManager;
+import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+import com.facebook.presto.spi.procedure.Procedure;
+import com.facebook.presto.spi.session.PropertyMetadata;
+import com.facebook.presto.spi.transaction.IsolationLevel;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.bootstrap.LifeCycleManager;
+import io.airlift.log.Logger;
+
+import javax.inject.Inject;
+
+import java.util.List;
+import java.util.Set;
+
+import static com.facebook.presto.spi.transaction.IsolationLevel.REPEATABLE_READ;
+import static com.facebook.presto.spi.transaction.IsolationLevel.checkConnectorSupports;
+import static java.util.Objects.requireNonNull;
+
+public class RaptorConnector
+        implements Connector
+{
+    private static final Logger log = Logger.get(RaptorConnector.class);
+
+    private final LifeCycleManager lifeCycleManager;
+    private final TransactionManager transactionManager;
+    private final NodeSupplier nodeSupplier;
+    private final BucketManager bucketManager;
+    private final TransactionWriter transactionWriter;
+    private final ConnectorSplitManager splitManager;
+    private final ConnectorPageSourceProvider pageSourceProvider;
+    private final ConnectorPageSinkProvider pageSinkProvider;
+    private final ConnectorNodePartitioningProvider nodePartitioningProvider;
+    private final List<PropertyMetadata<?>> sessionProperties;
+    private final List<PropertyMetadata<?>> tableProperties;
+    private final Set<Procedure> procedures;
+    private final Set<SystemTable> systemTables;
+
+    @Inject
+    public RaptorConnector(
+            LifeCycleManager lifeCycleManager,
```

Comment:
This seems like a good idea, but is problematic, since there's no easy way to inject it in Guice (requires custom object or annotation). I could create a closure in the constructor to avoid storing a full `LifeCycleManager`, but that seems more complicated for little benefit, and is different from other connectors.